### PR TITLE
DEPR: Index.__and__, __or__, __xor__ behaving as set ops

### LINKED ISSUE #36758: DEPR: |&^ as set operations for Index
It isn't that burdensome to write `left.union(right)` instead of `left | right`, and deprecating this special-case behavior allow us to fully share the implementation with Series.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #36758
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

xref #30228

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
can u link to the deprecation removal issue

also pls test the inplace ops produce a warning

eg

|= and so on

**Comment by jbrockmendel** [comment]
> can u link to the deprecation removal issue

done

> also pls test the inplace ops produce a warning

we dont support inplace ops for these

File: pandas/core/series.py [hunk 1]
```diff
@@ -725,7 +725,7 @@ def __array_ufunc__(
             # it to handle *args.
             index = alignable[0].index
             for s in alignable[1:]:
-                index |= s.index
```
Comment (jreback) [inline pandas/core/series.py:728]: we certainly are using this (or at least were), so need to make sure these inplace ops are deprecated as well.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/series.py (hunk 1 above)
Comment (jreback) [reply pandas/core/series.py:728]: can you add some tests for the inplace to make sure they are warning as well

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm merge on greenish

**Comment by jreback** [comment]
thanks @jbrockmendel

**Comment by jreback** [comment]
hmm this may have broken some tests: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=46712&view=logs&j=eab14f69-13b6-5db7-daeb-7b778629410b&t=ce687173-08c6-5301-838d-71b2dda24510

**Comment by jbrockmendel** [comment]
> hmm this may have broken some tests 

just opened #37587 for this

**Pull request merged**